whois-query/src/main/java/net/ripe/db/whois/query/acl/AccessControlListManager.java

```diff
@@ -65,13 +72,18 @@ public AccessControlListManager(final DateTimeProvider dateTimeProvider,
         this.ssoTokenTranslator = ssoTokenTranslator;
         this.ipRanges = ipRanges;
         this.isSSOAccountingEnabled = isSSOAccountingEnabled;
+        this.jdbcRpslObjectSlaveDao = jdbcRpslObjectSlaveDao;
     }
 
-    public boolean requiresAcl(final RpslObject rpslObject, final Source source) {
+    public boolean requiresAcl(final RpslObject rpslObject, final Source source, final AccountingIdentifier accountingIdentifier) {
         if (source.isGrs()) {
             return false;
         }
 
+        if (accountingIdentifier.getSsoUser() != null && isUserOwnedObject(rpslObject, accountingIdentifier.getSsoUser().uuid())){
```

Comment:
on a second thought , there is no need to check for isOwnedObject if objectType is not PERSON or Role. this check should go below the person or role check. In this case it makes sense to fetch the maintainers here only.

Reply:
I agree with this. I just refactored that method to check if it not a person or role beforehand